product_code/hyperledger/api/src/test/scala/ConnectionTests.scala

```diff
@@ -1,64 +1,62 @@
 import de.upb.cs.uc4.hyperledger.ConnectionManager
-import org.scalatest.FunSuite
+import org.scalatest.wordspec.AnyWordSpec
 import org.scalatest.PrivateMethodTester
 
-class ConnectionTests extends FunSuite with PrivateMethodTester {
-
-  /*  Simple Test to check for an available gateway according to the network configuration file
-   */
-  test("Check gateway connection") {
-    // retrieve possible identities
-    val wallet = ConnectionManager.getWallet()
-    assert(wallet != null, "Wallet retrieved was null.")
-
-    // prepare Network Builder
-    val builder = ConnectionManager.getBuilder(wallet)
-    assert(builder != null, "Builder retrieved was null, maybe the connection profile did not match the running network.")
-
-    // get gateway object
-    val gateway = builder.connect
-    assert(gateway != null, "Gateway retrieved was null.")
-
-    // cleanup
-    ConnectionManager.disposeGateway(gateway)
-  }
-
-  /*  Simple Test to check for an available connection to our network.
-   */
-  test("Check network connection") {
-
-    // retrieve possible identities
-    val wallet = ConnectionManager.getWallet()
-    assert(wallet != null, "Wallet retrieved was null.")
-
-    // prepare Network Builder
-    val builder = ConnectionManager.getBuilder(wallet)
-    assert(builder != null, "Builder retrieved was null, maybe the connection profile did not match the running network.")
-
-    // get gateway object
-    val gateway = builder.connect
-    assert(gateway != null, "Gateway retrieved was null.")
-
-    // try connecting to the network
-    try{
-      val network = gateway.getNetwork(ConnectionManager.channel_name)
-      assert(network != null, "Network retrieved was null.")
-    } finally {
-      ConnectionManager.disposeGateway(gateway)
+class ConnectionTests extends AnyWordSpec with PrivateMethodTester {
+
+  val connectionManager = ConnectionManager()
+
+  "The connectionManager" when {
+    "connecting to Chain" should {
+
+      "provide gateway connection" in {
+        // retrieve possible identities
+        val wallet = connectionManager.getWallet()
+        assert(wallet != null, "Wallet retrieved was null.")
+
+        // prepare Network Builder
+        val builder = connectionManager.getBuilder(wallet)
+        assert(builder != null, "Builder retrieved was null, maybe the connection profile did not match the running network.")

```

Comment:
We agreed to use Matchers with WordSpec for Testing

Reply:
fixed in [https://github.com/upb-uc4/University-Credits-4.0/pull/121/commits/710923f1f36afa50fc8faff88e18af10d9ceb896](https://github.com/upb-uc4/University-Credits-4.0/pull/121/commits/710923f1f36afa50fc8faff88e18af10d9ceb896)